src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -277,8 +277,34 @@ public override string ToString()
         }
     }
 
-    internal abstract class SafeFreeContextBufferChannelBinding : ChannelBinding
+    internal sealed class SafeFreeContextBufferChannelBinding : ChannelBinding
     {
-        // TODO (Issue #3362) To be implemented
+        private readonly SafeChannelBinding _channelBinding = null;
+
+        public override int Size
+        {
+            get { return _channelBinding.Length; }
+        }
+
+        public override bool IsInvalid
+        {
+            get { return _channelBinding.IsInvalid; }
+        }
+
+        public SafeFreeContextBufferChannelBinding(SafeChannelBinding binding)
+        {
+            Debug.Assert(null != binding && !binding.IsInvalid, "input channelBinding is invalid");
+            bool gotRef = false;
```

Comment:
Currently we are always doing DangerousRelease() in ReleaseHandle() irrespective of gotRef.
This is similar to SafeFreeCertContext implementation. Should we check gotRef in ReleaseHandle() ?
cc: @bartonjs @vijaykota